keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/Locatable.kt

```diff
@@ -30,7 +30,7 @@ interface Locations<T : RegionSpec> {
    * If not specified here, this should be derived from the [SubnetAwareLocations.subnet] (if
    * present) or use a default VPC name.
    */
-  val vpc: String
+  val vpc: String?
```

Comment:
Why have you made vpc nullable?

Reply:
Because it has a default and should be possible to omit from the spec, which was one of the specific requests from @erikmunson in the issue description (see #581). This is similar to other fields (like `subnet` in the implementation classes) which are already nullable. If it's not nullable, then I can't omit that value when exporting (or at least I didn't know of/find another way with the current Jackson config, or even modifying it).

Reply:
See https://github.com/spinnaker/keel/commit/bcf575aec24370c8b1dd78fcfa2e8b726daad87c#diff-570d68b946f65312a79f34d24cc10247 - vpc was only just changed from nullable to required.

Reply:
Hmm, I see. Thanks for the heads-up @asher. So what threw me off here is that, unlike other objects that show up in a cluster spec (say, a `ClusterSpec.LaunchConfigurationSpec` for the `launchConfiguration` field), the `locations` field just uses an implementation of `Locations`, which are the same classes used to represent the concrete resources after resolution...

What would you and @emjburns suggest here? Should I add a `ClusterSpec.LocationsSpec` so I can make those fields nullable? -- that seems like a much bigger change, though, I'm a bit reluctant. Or is there a better way of skipping those fields in the JSON output if they match certain values (custom deserializer?).

Reply:
@asher @emjburns Just checking back on this one. What would you advise on this chicken-and-egg problem with using the same data class for concrete resource definitions and for specs (where we want to make fields nullable for the export)?

Reply:
A `LocationSpec` where everything is nullable seems reasonable to me. @robfletcher any thoughts?

Reply:
Yeah, I guess so, although given we have multiple implementations of `Locations` already, are we going to need to double up the spec types as well?

Reply:
Right, which is why I was reluctant to do that in this PR. Unless there are major objections, and given that truly `vpc` is intended to be optional and that we currently use those classes the way we do, I'd propose we keep it nullable for now and revisit this in a separate PR.

Reply:
Yeah, it's a tough call because it genuinely is nullable in the spec but mandatory in the hydrated resource model. Thus far we've created separate classes when that is the case.

Reply:
Not in this case. 😄

Reply:
@robfletcher I started down the path of creating a new data class for `SubnetAwareLocationsSpec`, but it's gonna have to implement the `Locations` interface so that `ClusterSpec` can still be `Locatable`, and changing that type in `ClusterSpec` has a ripple effect that touches a ton of places in the code. Plus, I would still need to make the `vpc` field in `Locations` nullable since, well, it can be `null` in specs...

Can we agree on keeping it nullable across the existing interfaces/classes for now and revisit this in a separate PR?